### PR TITLE
MM-47087 : Migrate "components/integrations/integrations.jsx"

### PR DESCRIPTION
**Pull request by babinderrathi**
#### Summary
Migration of integrations.jsx to integrations.tsx

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/21082

  Fixes https://mattermost.atlassian.net/browse/MM-47087

#### Related Pull Requests
None 

#### Screenshots
None

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by AshishDhama — DISMISSED** [review]

File: components/integrations/index.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import {connect} from 'react-redux';
 
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 
-import Integrations from './integrations.jsx';
+import Integrations from './integrations';
```
Comment (M-ZubairAhmed) [inline components/integrations/index.js:8]: this needs to be converted to "ts"

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: components/integrations/integrations.tsx [hunk 2]
```diff
@@ -20,19 +19,21 @@ import TeamPermissionGate from 'components/permissions_gates/team_permission_gat
...
+    siteName: string;
+    enableIncomingWebhooks: string;
+    enableOutgoingWebhooks: string;
+    enableCommands: string;
+    enableOAuthServiceProvider: string;
+    team: {
+        name: string;
+        id: string;
+        display_name: string;
+    };
```
Comment (M-ZubairAhmed) [inline components/integrations/integrations.tsx:32]: this needs to be `Team` type

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by nevyangelova — CHANGES_REQUESTED** [review]
Thanks @babinderrathi would it be possible to convert the index.js file as well?

Also I can see a few places in the component functions that could be typed as well. Do you mind double checking?

**Review by nevyangelova — APPROVED** [review]

File: components/integrations/integrations.tsx (hunk 2 above)
Comment (M-ZubairAhmed) [reply components/integrations/integrations.tsx:32]: @babinderrathi please take a look at this one

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]
💯 It looks good to me. Thank you 🎉

**Pull request merged**